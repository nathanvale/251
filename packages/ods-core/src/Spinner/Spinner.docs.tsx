import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Text } from "../Text";
import { Columns } from "../Columns";
import { Column } from "../Column";
import { Box } from "../Box";
import { Spinner, SpinnerProps, SpinnerSize } from "./Spinner";

export const docs: ComponentDocs<SpinnerProps> = {
  category: "Atomic",
  componentName: "Spinner",
  description: [
    "This component is used to show a spinner while the user is waiting for an action to be done.",
  ].join(" "),
  propDescriptions: {
    size: [
      `Size of the Spinner.`,
      `The "xsmall" size is for very specific use cases. Do not use this size unless you know what you are doing.`,
    ].join(" "),
    tone: [`Available colours to choose for the Spinner.`].join(" "),
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => <Spinner />,
    },
    additional: [
      {
        label: "Inherit tone form parent context",
        description: [
          `You can change the tone to inherit from color value of its parent.`,
        ].join(" "),
        Code: () => (
          <Text tone="positive">
            This is inherited colour <Spinner tone="inherit" />
          </Text>
        ),
      },
      {
        label: "Tone white, on dark backgrounds",
        description: `If the background is dark, use the "white" tone for spinner.`,
        playroom: false,
        Code: () => (
          <Box
            display="flex"
            alignItems="center"
            justifyContent="center"
            style={{ backgroundColor: "rgb(236, 0, 0)", height: "100px" }}
          >
            <Spinner tone="white" />
          </Box>
        ),
      },
      {
        label: "Different spinner sizes",
        Code: () => (
          <Columns space="medium">
            {(["xsmall", "small", "medium", "large"] as SpinnerSize[]).map(
              (size) => (
                <Column key={size}>
                  <Spinner size={size} />
                </Column>
              )
            )}
          </Columns>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Basic loading",
      Code: () => <Spinner />,
    },
    {
      label: "with tone and size",
      Code: () => <Spinner size="large" tone="inherit" />,
    },
  ],
};
