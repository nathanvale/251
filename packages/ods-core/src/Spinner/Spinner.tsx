import * as React from "react";
import { CircularProgress } from "@material-ui/core";
import { ComponentBaseProps } from "@origin-digital/ods-types";

export interface SpinnerProps extends ComponentBaseProps {
  size?: "xsmall" | "small" | "medium" | "large";
  color?: "inherit" | "primary" | "secondary";
}

export const Spinner = ({
  color = "secondary",
  size = "xsmall",
  ...other
}: SpinnerProps) => {
  let spinnerSize = 20;
  if (size === "small") {
    spinnerSize = 24;
  } else if (size === "medium") {
    spinnerSize = 48;
  } else if (size === "large") {
    spinnerSize = 72;
  }

  return <CircularProgress color={color} size={spinnerSize} {...other} />;
};

Spinner.displayName = "Spinner";
Spinner.defaultProps = {
  color: "secondary",
  "data-id": "spinner",
  size: "small",
};
