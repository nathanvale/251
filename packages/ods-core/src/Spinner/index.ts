export { Spinner } from "./Spinner";
export type SpinnerProps = import("./Spinner").SpinnerProps;
