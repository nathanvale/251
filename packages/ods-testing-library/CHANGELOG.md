# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
