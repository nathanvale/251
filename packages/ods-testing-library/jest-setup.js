import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
