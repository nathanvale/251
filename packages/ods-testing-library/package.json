{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
