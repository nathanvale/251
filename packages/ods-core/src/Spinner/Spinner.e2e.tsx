import * as React from "react";
import { E2ETests } from "@origin-digital/ods-types";
import { Box } from "../Box/Box";
import { Column } from "../Column/Column";
import { Columns } from "../Columns/Columns";
import { Spinner, SpinnerSize, SpinnerColor } from "./Spinner";

export const tests: E2ETests = [
  {
    label: "Default Spinner",
    Code: () => <Spinner />,
  },
  {
    label: "Spinner in different colors",
    Code: () => (
      <Columns space="medium">
        {(["inherit", "secondary", "white"] as SpinnerColor[]).map((color) => (
          <Column key={color}>
            {color === "white" ? (
              <Box
                display="flex"
                alignItems="center"
                justifyContent="center"
                style={{ backgroundColor: "rgb(236, 0, 0)", height: "100px" }}
              >
                <Spinner color={color} />
              </Box>
            ) : (
              <Spinner color={color} />
            )}
          </Column>
        ))}
      </Columns>
    ),
  },
  {
    label: "Spinner in different sizes",
    Code: () => (
      <Columns space="medium">
        {(["xsmall", "small", "medium", "large"] as SpinnerSize[]).map(
          (size) => (
            <Column key={size}>
              <Spinner size={size} />
            </Column>
          )
        )}
      </Columns>
    ),
  },
];
