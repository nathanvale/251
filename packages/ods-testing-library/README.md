# @origin-digital/ods-testing-library

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Installation](#installation)
- [Why do I need this package?](#why-do-i-need-this-package)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

This package enhances the render function of `react-testing-library` so that the `<OriginThemeProvider/>` is wrapped around your rendered component.

All other API features of `react-testing-library` are also included.

## Installation

Install the package in your project directory with:

```sh
yarn add @origin-digital/ods-testing-library
```

This package also has a peer dependency on the following:

```sh
// with yarn
yarn add @origin-digital/ods-core
yarn add -D @testing-library/react
yarn add -D jest-styled-components
```

## Why do I need this package?

**The problem**

When using react-testing-library and ODS with the jest testing framework it is necessary to wrap **all** your components in our `<OriginThemeProvider/>` - otherwise the react component cannot be rendered and tested.

**The solution**

`@origin-digital/ods-testing-library` creates a wrapper around the `react-testing-library` render function. It is here that we wrap the `<OriginThemeProvider/>`. All other API features of `react-testing-library` are also included.

This practice is recommended by the `react-testing-library` documentation itself...

https://testing-library.com/docs/native-testing-library/setup#custom-render

The origin-ui dogfoods this package itself but it can also be used in your own app to test by adding to your jest config:

```
{
...,
setupFilesAfterEnv: [@origin-digital/ods-testing-library/jest-setup],
...
}
```

and usage in your tests:

```
import React from 'react';
import {render} from '@origin-digital/ods-testing-library';
import {Box} from '../Box/Box';
import {Stack} from './Stack';

test('It can stack', () => {
  const {container} = render(
    <Stack>
      <Box backgroundColor="red" padding="medium" />
      <Box backgroundColor="orange" padding="medium" />
      <Box backgroundColor="blue" padding="medium" />
    </Stack>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
```
