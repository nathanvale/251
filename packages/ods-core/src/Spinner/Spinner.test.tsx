import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Spinner, SpinnerProps } from "./Spinner";

const renderSpinner = (spinnerProps: Partial<SpinnerProps> = {}) =>
  render(<Spinner {...spinnerProps} />);

test("It renders in default", () => {
  const { container } = renderSpinner();
  expect(container).toMatchSnapshot();
});
