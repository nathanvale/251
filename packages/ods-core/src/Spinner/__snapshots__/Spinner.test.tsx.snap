// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders in default 1`] = `
<div>
  <div
    class="MuiCircularProgress-root spinner-root-1 spinner-root-2 MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
    data-id="spinner"
    role="progressbar"
    style="width: 24px; height: 24px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;
